wasi/README.md

```diff
@@ -0,0 +1,105 @@
+aimed at `preview1`.
+
+> **Note**: You may hear many names for referencing these versions. You might hear the terms `preview1/preview1` or `0.1/0.2` when referring to the versions inside the WASI docs.
+> And you might hear `wasip1` when being used as a flag to a compiler target. We tend to prefer the nomenclature wasip1 / wasip2.
+
+### wasip1
+
+Although `wasip1` is marked as "legacy", this is the version that nearly all compilers support when compiling to a Wasm target. For that reason we are aiming to have good support, but have no immediate plans to "complete" the implementation.
+Please reach out if you'd like to see more done to support wasip1.
+
+See the [WasiPreview1.java](https://github.com/dylibso/chicory/blob/main/wasi/src/main/java/com/dylibso/chicory/wasi/WasiPreview1.java) class
+to understand what we support today and what we do not support. Here are some features have basic coverage for currently:
+
+* stdin / stdout / stderr
+* environment variables
+* command arguments
+* clocks
+* random
+* basic reading and writing of files (through use of a [virtual file system](https://github.com/google/jimfs))
+
+If your module calls a wasi function that we don't support, or uses a feature that we don't support, we will throw a `WASMRuntimeException`.
+
+### wasip2
+
+We do have intentions to support wasip2 in the future, however this work has not been started. Please reach out to us on zulip if you are interested in helping plan and execute this work.
+
+## How to use
+
+As a host who is running Wasm modules, WASI is just a collection of host imports that you need to provide
+to a wasi-compiled module when instantiating it. You'll also need to configure some options for how
+these functions behave and what the module can and cannot do.
+
+### Bare-Bones Instantiation
+
+So to instantiate a WASI module you need an instance of `WasiPreview1`. You can turn this instance into
+import functions which can then be passed to the Module builder. 
+
+```java
+var logger = new SystemLogger();
+// let's just use the default options for now
+var options = WasiOptions.builder().build();
+// create our instance of wasip1
+var wasi = new WasiPreview1(this.logger, WasiOptions.builder().build());
+// turn those into host imports. Here we could add any other custom imports we have
+var imports = new HostImports(wasi.toHostFunctions());
+// create the module
+var module = Module.builder("hello-wasi.wasm").build();
+// instantiate and connect our imports, this will execute the module
+module.instantiate(imports);
+```
+
+> **Note**: Take note that we don't explicitly execute the module. The module will run when you instantiate it. This
+> is part of the WASI spec. They will implicitly call `_start`. To learn more [read this blog post](https://dylibso.com/blog/wasi-command-reactor/).
```

Comment:
Also a link to the spec? https://webassembly.github.io/spec/core/syntax/modules.html#start-function